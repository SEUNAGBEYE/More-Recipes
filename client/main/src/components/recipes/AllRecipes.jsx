import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  getFavouritedRecipesIds,
  addRecipe,
  allRecipes,
  recipeCategories
}
  from '../../actions/Recipes';
import RecipeCard from './RecipeCard';
import Pagination from './Pagination';
import Exclamation from './Exclamation';
import RecipeModal from './RecipeModal';
import CategoryButton from './CategoryButton';

/**
 * @export
 * @class AllRecipes
 * @extends {Component}
 */
export class AllRecipes extends Component {
  /**
   * Creates an instance of AllRecipes.
   * @param {Object} props
   * @memberof AllRecipes
   */
  constructor(props) {
    super(props);
    this.onSubmit = this.onSubmit.bind(this);
    this.paginateRecipes = this.paginateRecipes.bind(this);
  }

  /**
   * @returns {void} void
 * @memberof AllRecipes
 */
  componentDidMount() {
    this.props.recipeCategories();
    this.paginateRecipes(1);
  }

  /**
   * @param {Number} page
   *
   * @returns {void} void
   * @memberof AllRecipes
   */
  paginateRecipes(page) {
    this.props.allRecipes(page);
  }


  /**
   * @param {Object} data
   *
   * @returns {void} void
   * @memberof AllRecipes
   */
  onSubmit(data) {
    this.props.addRecipe(data)
      .then(res => {
        this.setState({ allRecipes: [...this.state.allRecipes, res.recipe] });
      });
  }

  /**
   * @memberOf UserRecipes
   * @returns {jsx} JSX
   * return {object}
   */
  render() {
    return (
      <div>
        <main style={{ marginTop: 40 }}>

          <div className="container">
            <CategoryButton/>
          </div>

          <div className="container">
            <div style={{ textAlign: 'center', marginTop: 100 }}>
              <h4 className="container__myrecipes">All Recipes</h4><br /><br />
              <RecipeModal
                addRecipe={this.onSubmit}
                recipeCategories={this.props.categories}
              />
            </div>

            <div className="row">
              {
                this.props.recipes.length > 0 ?
                  this.props.recipes.map((recipe, index) => (
                    <RecipeCard key={recipe.id}
                      recipe={recipe}
                      id={recipe.id}
                      history={this.props.history}
                    />)) :
                  <Exclamation>
                    <p className="text-muted text-center">
                      Sorry no recipe has been added yet,
                      please add to get started
                    </p>
                  </Exclamation>
              }
            </div>
          </div>
        </main>
        {this.props.pagination > 1 ? <Pagination recipesCount={this.props.pagination}
          recipesPagination={this.paginateRecipes}/> :
          ''
        }
      </div>
    );
  }
}

const propTypes = {
  recipes: PropTypes.array.isRequired,
  addRecipe: PropTypes.func.isRequired,
  recipeCategories: PropTypes.func.isRequired,
  allRecipes: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
  pagination: PropTypes.number.isRequired,
  history: PropTypes.object.isRequired,
  user: PropTypes.object.isRequired
};

AllRecipes.propTypes = propTypes;

/**
 * mapStateToProps
 * @export
 * @param {Object} state
 *
 * @returns {Object} object
 */
export const mapStateToProps = (state) => ({
  recipes: state.recipes.allRecipes,
  pagination: Number(state.recipes.pagination),
  user: state.auth.user,
  isAuthenticated: state.auth.isAuthenticated,
  categories: state.recipes.recipeCategories
});

export default connect(mapStateToProps, {
  getFavouritedRecipesIds,
  allRecipes,
  addRecipe,
  recipeCategories
})(AllRecipes);

