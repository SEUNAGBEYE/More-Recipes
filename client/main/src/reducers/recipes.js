import actionTypes from '../actions/actionTypes';

const {
  GET_RECIPE,
  GET_RECIPES,
  GET_USER_RECIPES,
  GET_USER_FAVOURITED_RECIPES_ID,
  GET_FAVOURITED_RECIPES,
  GET_POPULAR_RECIPES,
  GET_RECIPE_REVIEWS,
  RECIPE_CATEGORIES,
  SEARCH_RECIPES,
  ADD_RECIPE,
  DELETE_RECIPE,
  EDIT_RECIPE,
  TOGGLE_FAVOURITE_RECIPE,
  TOGGLE_THUMBS_UP_RECIPE,
  TOGGLE_THUMBS_DOWN_RECIPE,
  REVIEW_RECIPE
} = actionTypes;

export const initialState = {
  allRecipes: [],
  singleRecipe: {
    id: '',
    name: '',
    image: '',
    categoryId: '',
    userId: '',
    upvotes: [],
    downvotes: [],
    views: [],
    description: '',
    reviews: []
  },
  pagination: '',
  userFavouritedRecipeId: [],
  favouriteRecipes: [],
  popularRecipes: [],
  recipeCategories: [],
  loaded: false
};

export default (state = initialState, action = {}) => {
  const singleRecipe = { ...state.singleRecipe };
  let allRecipes = [...state.allRecipes];
  const { recipe: recipeOnAction } = action;

  switch (action.type) {
  case ADD_RECIPE:
    return {
      ...state,
      ...{ allRecipes: [action.recipe, ...state.allRecipes] }
    };

  case GET_RECIPE:
    return {
      ...state,
      ...{
        singleRecipe: action.recipe,
        pagination: action.pagination,
        loaded: true
      }
    };

  case GET_USER_RECIPES:
    return {
      ...state,
      ...{
        allRecipes: action.recipes,
        pagination: action.pagination
      }
    };

  case GET_RECIPES:
    return {
      ...state,
      ...{
        allRecipes: action.recipes,
        pagination: action.pagination,
        loaded: true
      }
    };

  case RECIPE_CATEGORIES:
    return {
      ...state,
      ...{
        recipeCategories: action.recipeCategories,
        loaded: true
      }
    };
  case GET_POPULAR_RECIPES:
    return {
      ...state,
      ...{
        popularRecipes: [...action.popularRecipe],
        loaded: true
      }
    };

  case SEARCH_RECIPES:
    return {
      ...state,
      ...{
        allRecipes: [...action.recipes],
        pagination: action.pagination
      }
    };

  case DELETE_RECIPE:
    return {
      ...state,
      ...{
        allRecipes: state.allRecipes
          .filter(recipe => parseInt(recipe.id, 10) !== parseInt(action.id, 10))
      }
    };


  case EDIT_RECIPE:
    if (allRecipes.length >= 1) {
      allRecipes = allRecipes.map((recipe) => {
        if (parseInt(recipe.id, 10) === parseInt(action.recipe.id, 10)) {
          return action.recipe;
        }
        return recipe;
      });
    }

    return {
      ...state,
      ...{
        allRecipes,
        singleRecipe: { ...singleRecipe, ...recipeOnAction }
      }
    };

  case TOGGLE_FAVOURITE_RECIPE:
    return {
      ...state,
      ...{
        favouriteRecipes: !state.userFavouritedRecipeId
          .includes(action.favouritedRecipe.id) ?
          [...state.favouriteRecipes, action.favouritedRecipe] :
          state.favouriteRecipes
            .filter(recipe => recipe.id != action.favouritedRecipe.id),

        userFavouritedRecipeId: !state.userFavouritedRecipeId
          .includes(action.favouritedRecipe.id) ?
          state.userFavouritedRecipeId.concat(action.favouritedRecipe.id) :
          state.userFavouritedRecipeId
            .filter(id => id !== action.favouritedRecipe.id),
        singleRecipe: action.favouritedRecipe
      }
    };

  case TOGGLE_THUMBS_UP_RECIPE:
    return {
      ...state,
      ...{
        allRecipes: [...state.allRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        favouriteRecipes: [...state.favouriteRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        singleRecipe: action.recipe
      }
    };

  case TOGGLE_THUMBS_DOWN_RECIPE:
    return {
      ...state,
      ...{
        allRecipes: [...state.allRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        favouriteRecipes: [...state.favouriteRecipes.map((recipe) => {
          if (recipe.id === action.recipe.id) {
            return action.recipe;
          }
          return recipe;
        })],
        singleRecipe: action.recipe
      }
    };

  case GET_FAVOURITED_RECIPES:
    return {
      ...state,
      favouriteRecipes: [...action.favouriteRecipes],
      pagination: action.pagination
    };

  case GET_USER_FAVOURITED_RECIPES_ID:
    return {
      ...state,
      ...{ userFavouritedRecipeId: action.favouritedRecipesIds }
    };
  case REVIEW_RECIPE:
    singleRecipe.reviews = [action.reviewBody, ...singleRecipe.reviews];
    return {
      ...state,
      ...{ singleRecipe }
    };
  case GET_RECIPE_REVIEWS:
    singleRecipe.reviews = [...singleRecipe.reviews, ...action.reviews];
    return {
      ...state,
      ...{
        singleRecipe,
        pagination: action.pagination
      }
    };

  default: return {
    ...state,
    ...{ loaded: true }
  };
  }
};

export { actionTypes };
