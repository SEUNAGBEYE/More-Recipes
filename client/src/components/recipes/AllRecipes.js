import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Navbar from '../common/Navbar';
import Footer from '../common/Footer';
import { getFavouritedRecipesIds, makeFavouriteRecipe, addRecipe, getUserRecipes, allRecipes, favouriteRecipe, deleteRecipe, editRecipe} from '../../actions/Recipes';
import RecipeCard from './RecipeCard';
import Pagination from './Pagination';
import Exclamation from './Exclamation';
import RecipeModal from './RecipeModal';



/**
 * @class UserRecipes
 */
class AllRecipes extends Component{

	constructor(props){
		super(props);
		this.state = {
			name: '',
			description: '',
			image: 'hello image',
			ingredients: [],
			steps: [],
      errors: {},
      categoryId: '',
      downvotes: [],
      upvotes:[],
      allRecipes: [],
      favouritedRecipesIds: []
		}
    this.onSubmit = this.onSubmit.bind(this);
    this.deleteRecipe = this.deleteRecipe.bind(this);
    this.editRecipe = this.editRecipe.bind(this);
    this.makeFavouriteRecipe = this.makeFavouriteRecipe.bind(this)
  }
  

  componentDidMount(){
    this.props.allRecipes()
    .then(res => {
      this.setState({allRecipes: [...this.state.allRecipes, ...res.allRecipes]})
    });

    // console.log('component Mounted')
    // this.props.getFavouritedRecipesIds()
    // .then(res => {
		// 	console.log('component Mounted for me', res)
    //   this.setState({favouritedRecipeId: [...res]})
    // })
  }
  
  // componenDidUpdate(){
  //   getFavouritedRecipesIds()
  //   .then(res => {
	// 		console.log('component updated', res)
  //     this.setState({favouritedRecipesIds: [...res]})
  //   })
  // }

  onSubmit(data){
    this.props.addRecipe(data)
    .then(res => {
      this.setState({allRecipes: [...this.state.allRecipes, res.recipe]})
    })
  }

  deleteRecipe(id){
    this.props.deleteRecipe(id)
    .then(res => {
      this.setState({allRecipes: this.state.allRecipes.filter( recipe => recipe.id !== id) })
    })
  }

	makeFavouriteRecipe(id){
		makeFavouriteRecipe(id)
		.then(res => {
			console.log('alldata',res.data.favouritedRecipesId
    )
      this.setState({favouritedRecipesIds: [...res.data.favouritedRecipesId]})
    })
	}

  editRecipe(id, recipe){
    this.props.editRecipe(id, recipe)
    .then(res => {
      this.setState({allRecipes: [...this.state.allRecipes.filter( recipe => recipe.id === id)] })
    })
  }


  /**
   * @memberOf UserRecipes
   * return {object}
   */
  render(){
    return(
      <div>
        <Navbar />
        <main style={{marginTop: 40}}>

          <div className="container">
            <div className="dropdown" style={{float: 'left'}}>
              <button className="btn btn-default dropdown-toggle auth-button" type="button" id="about-us" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Category
              </button>
              <div className="dropdown-menu" aria-labelledby="about-us">
                <a className="dropdown-item" href="recipes.html">Dessert</a>
                <a className="dropdown-item" href="my_recipes.html">Pasta</a>
                <a className="dropdown-item" href="favourite_recipe.html">Fries</a>
                <a className="dropdown-item" href="#">Chinese</a>
                <a className="dropdown-item" href="index.html">Africa</a>
              </div>
            </div>
          </div>
          
          <div className="container">
            <div style={{textAlign: 'center', marginTop: 100}}>
              <h4 className='container__myrecipes'>All Recipes</h4><br /><br />
              <RecipeModal addRecipe={this.onSubmit}/>
            </div>

            <div className='row'>
              {
                this.state.allRecipes.length > 0
                ?
                this.state.allRecipes.reverse().map((elem, index) => {
                return (<RecipeCard key={index} 
                user={this.props.user}
                recipe={elem}
                id={elem.id} 
                onDelete={this.deleteRecipe} 
                editRecipe={this.editRecipe} 
                makeFavouriteRecipe={this.makeFavouriteRecipe}
                />)
                })
                :
                <Exclamation>
                  <p className='text-muted text-center'>Sorry no recipe has been added yet, please add to get started</p>
                </Exclamation>
              }
            </div>
          </div>
        </main>
        <Pagination />
        <Footer />
      </div>
    )
  }
}

/**
 * mapStateToProps
 * @param {any} state
 * @returns {object} object
 */
const mapStateToProps = (state) => {
  return {
    recipes: state.recipes,
    user: state.auth.user
  };
}

export default connect(mapStateToProps, {getFavouritedRecipesIds, makeFavouriteRecipe, allRecipes, addRecipe, getUserRecipes, deleteRecipe, editRecipe })(AllRecipes);

