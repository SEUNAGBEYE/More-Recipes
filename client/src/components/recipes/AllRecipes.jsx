import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { getFavouritedRecipesIds, toggleFavouriteRecipe, addRecipe, getUserRecipes, allRecipes, favouriteRecipe, deleteRecipe, editRecipe } from '../../actions/Recipes';
import RecipeCard from './RecipeCard';
import Pagination from './Pagination';
import Exclamation from './Exclamation';
import RecipeModal from './RecipeModal';


/**
 * @class AllRecipes
 * @extends {Component}
 */
class AllRecipes extends Component {
  /**
   * Creates an instance of AllRecipes.
   * @param {any} props
   * @memberof AllRecipes
   */
  constructor(props) {
    super(props);

    this.onSubmit = this.onSubmit.bind(this);
    this.deleteRecipe = this.deleteRecipe.bind(this);
    this.editRecipe = this.editRecipe.bind(this);
    this.toggleFavouriteRecipe = this.toggleFavouriteRecipe.bind(this);
    this.paginateRecipes = this.paginateRecipes.bind(this);
  }

  /**
   * @returns {void} void
 * @memberof AllRecipes
 */
  componentDidMount() {
    this.paginateRecipes(1);

    this.props.getFavouritedRecipesIds()
      .then(res => {
        this.setState({ favouritedRecipeIds: [...res.favouritedRecipesIds] });
      });
  }

  /**
   * @returns {void} void
   * @param {any} page
   * @memberof AllRecipes
   */
  paginateRecipes(page) {
    this.props.allRecipes(page)
      .then(res => {
        this.setState({ allRecipes: [...res.allRecipes] });
      });
  }


  /**
   * @param {any} data
   * @returns {void} void
   * @memberof AllRecipes
   */
  onSubmit(data) {
    this.props.addRecipe(data)
      .then(res => {
        this.setState({ allRecipes: [...this.state.allRecipes, res.recipe] });
      });
  }

  /**
   * @param {any} id
   * @returns {void} void
   * @memberof AllRecipes
   */
  deleteRecipe(id) {
    this.props.deleteRecipe(id)
      .then(res => {
        this.setState({ allRecipes: this.state.allRecipes.filter(recipe => recipe.id !== id) });
      });
  }

  /**
 * @param {any} id
 * @returns {void} void
 * @memberof AllRecipes
 */
  toggleFavouriteRecipe(id) {
    toggleFavouriteRecipe(id)
      .then(res => {
        this.setState({ favouritedRecipesIds: [...res.data.favouritedRecipesId] });
      });
  }


  /**
   * @param {any} id
   * @returns {void} void
   * @param {any} recipe
   * @memberof AllRecipes
   */
  editRecipe(id, recipe) {
    this.props.editRecipe(id, recipe)
      .then(res => {
        this.setState({ allRecipes: [...this.state.allRecipes.filter(recipe => recipe.id === id)] });
      });
  }


  /**
   * @memberOf UserRecipes
   * @returns {jsx} JSX
   * return {object}
   */
  render() {
    return (
      <div>
        <main style={{ marginTop: 40 }}>

          <div className="container">
            <div className="dropdown" style={{ float: 'left' }}>
              <button className="btn btn-default dropdown-toggle auth-button" type="button" id="about-us" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">Category
              </button>
              <div className="dropdown-menu" aria-labelledby="about-us">
                <Link className="dropdown-item" to="recipes.html">Dessert</Link>
                <Link className="dropdown-item" to="my_recipes.html">Pasta</Link>
                <Link className="dropdown-item" to="favourite_recipe.html">Fries</Link>
                <Link className="dropdown-item" to="#">Chinese</Link>
                <Link className="dropdown-item" to="index.html">Africa</Link>
              </div>
            </div>
          </div>

          <div className="container">
            <div style={{ textAlign: 'center', marginTop: 100 }}>
              <h4 className="container__myrecipes">All Recipes</h4><br /><br />
              <RecipeModal addRecipe={this.onSubmit}/>
            </div>

            <div className="row">
              {
                this.props.recipes.length > 0 ?
                  this.props.recipes.map((elem, index) => (<RecipeCard key={elem.id}
                    user={this.props.user}
                    recipe={elem}
                    id={elem.id}
                    onDelete={this.deleteRecipe}
                    editRecipe={this.editRecipe}
                    toggleFavouriteRecipe={this.toggleFavouriteRecipe}
                    history={this.props.history}
                  />)) :
                  <Exclamation>
                    <p className="text-muted text-center">Sorry no recipe has been added yet, please add to get started</p>
                  </Exclamation>
              }
            </div>
          </div>
        </main>
        <Pagination recipesCount={this.props.pagination} recipesPagination={this.paginateRecipes}/>
      </div>
    );
  }
}

/**
 * mapStateToProps
 * @param {any} state
 * @returns {object} object
 */
const mapStateToProps = (state) => ({
  recipes: state.recipes.allRecipes,
  pagination: state.recipes.pagination,
  user: state.auth.user
});

export default connect(mapStateToProps, {
  getFavouritedRecipesIds, toggleFavouriteRecipe, allRecipes, addRecipe, getUserRecipes, deleteRecipe, editRecipe
})(AllRecipes);

