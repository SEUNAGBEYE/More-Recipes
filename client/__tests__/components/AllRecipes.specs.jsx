
import React from 'react';
import { shallow } from 'enzyme';
import { AllRecipes, mapStateToProps } from '../../main/src/components/recipes/AllRecipes';

const state = {
  auth: {
    user: {
      firstName: 'Seun',
      lastName: 'Agbeye'
    }
  },
  recipes: {
    recipeCategories: [{
      id: 1,
      name: 'Breakfast'
    }],
    popularRecipes: [{
      id: 1,
      name: 'Breakfast'
    }],
    allRecipes: [{
      id: 1,
      name: 'Breakfast'
    }],
  }
};

const recipe = {
  name: 'Amala',
  image: 'This is the image'
}


const props = {
  recipes: [],
  pagination: 1,
  addRecipe: (() =>new Promise((resolve, reject) => {
    recipe ? resolve(recipe) : reject('Recipe not found');
  })),
  allRecipes: jest.fn(),
  recipeCategories: jest.fn(),
  categories: [],
  history: {
    push: (() => {})
  },
  categories: [],
  user: {
    firstName: 'Seun',
    lastName: 'Agbeye'
  }
};

const event = {
  preventDefault: jest.fn(),
  target: {
    search: {
      value: '',
    },
    id: ''
  }
}

describe('# AllRecipes', () => {
  const wrapper = shallow(
      <AllRecipes {...props} />
  );
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    expect(wrapper.getElement().type).toBe('div');
    expect(wrapper).toMatchSnapshot()
    done();
  });
  it('should call onSumbmit', (done) => {
    const onSubmitSpy = jest.spyOn(wrapper.instance(), 'onSubmit')
    wrapper.instance().onSubmit(recipe);
    expect(onSubmitSpy).toHaveBeenCalledTimes(1)
    done();
  });

  it('should call mapStateToProps(state)', (done) => {
    mapStateToProps(state)
    done();
  }); 
});
