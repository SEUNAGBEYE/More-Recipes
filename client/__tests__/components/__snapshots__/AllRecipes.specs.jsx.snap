// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# AllRecipes should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AllRecipes
    addRecipe={[Function]}
    allRecipes={
      [MockFunction] {
        "calls": Array [
          Array [
            1,
          ],
        ],
      }
    }
    categories={Array []}
    history={
      Object {
        "push": [Function],
      }
    }
    pagination={1}
    recipeCategories={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    recipes={Array []}
    user={
      Object {
        "firstName": "Seun",
        "lastName": "Agbeye",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <main
          style={
            Object {
              "marginTop": 40,
            }
          }
        >
          <div
            className="container"
          >
            <Connect(CategoryButton) />
          </div>
          <div
            className="container"
          >
            <div
              style={
                Object {
                  "marginTop": 100,
                  "textAlign": "center",
                }
              }
            >
              <h4
                className="container__myrecipes"
              >
                All Recipes
              </h4>
              <br />
              <br />
              <RecipeModal
                addRecipe={[Function]}
                recipeCategories={Array []}
              />
            </div>
            <div
              className="row"
            >
              <Exclamation>
                <p
                  className="text-muted text-center"
                >
                  Sorry no recipe has been added yet, please add to get started
                </p>
              </Exclamation>
            </div>
          </div>
        </main>,
        <Pagination
          recipesCount={1}
          recipesPagination={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="container"
            >
              <Connect(CategoryButton) />
            </div>,
            <div
              className="container"
            >
              <div
                style={
                  Object {
                    "marginTop": 100,
                    "textAlign": "center",
                  }
                }
              >
                <h4
                  className="container__myrecipes"
                >
                  All Recipes
                </h4>
                <br />
                <br />
                <RecipeModal
                  addRecipe={[Function]}
                  recipeCategories={Array []}
                />
              </div>
              <div
                className="row"
              >
                <Exclamation>
                  <p
                    className="text-muted text-center"
                  >
                    Sorry no recipe has been added yet, please add to get started
                  </p>
                </Exclamation>
              </div>
            </div>,
          ],
          "style": Object {
            "marginTop": 40,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(CategoryButton) />,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  style={
                    Object {
                      "marginTop": 100,
                      "textAlign": "center",
                    }
                  }
                >
                  <h4
                    className="container__myrecipes"
                  >
                    All Recipes
                  </h4>
                  <br />
                  <br />
                  <RecipeModal
                    addRecipe={[Function]}
                    recipeCategories={Array []}
                  />
                </div>,
                <div
                  className="row"
                >
                  <Exclamation>
                    <p
                      className="text-muted text-center"
                    >
                      Sorry no recipe has been added yet, please add to get started
                    </p>
                  </Exclamation>
                </div>,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h4
                      className="container__myrecipes"
                    >
                      All Recipes
                    </h4>,
                    <br />,
                    <br />,
                    <RecipeModal
                      addRecipe={[Function]}
                      recipeCategories={Array []}
                    />,
                  ],
                  "style": Object {
                    "marginTop": 100,
                    "textAlign": "center",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "All Recipes",
                      "className": "container__myrecipes",
                    },
                    "ref": null,
                    "rendered": "All Recipes",
                    "type": "h4",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "br",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "addRecipe": [Function],
                      "recipeCategories": Array [],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Exclamation>
                    <p
                      className="text-muted text-center"
                    >
                      Sorry no recipe has been added yet, please add to get started
                    </p>
                  </Exclamation>,
                  "className": "row",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": <p
                      className="text-muted text-center"
                    >
                      Sorry no recipe has been added yet, please add to get started
                    </p>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Sorry no recipe has been added yet, please add to get started",
                      "className": "text-muted text-center",
                    },
                    "ref": null,
                    "rendered": "Sorry no recipe has been added yet, please add to get started",
                    "type": "p",
                  },
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "recipesCount": 1,
          "recipesPagination": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <main
            style={
              Object {
                "marginTop": 40,
              }
            }
          >
            <div
              className="container"
            >
              <Connect(CategoryButton) />
            </div>
            <div
              className="container"
            >
              <div
                style={
                  Object {
                    "marginTop": 100,
                    "textAlign": "center",
                  }
                }
              >
                <h4
                  className="container__myrecipes"
                >
                  All Recipes
                </h4>
                <br />
                <br />
                <RecipeModal
                  addRecipe={[Function]}
                  recipeCategories={Array []}
                />
              </div>
              <div
                className="row"
              >
                <Exclamation>
                  <p
                    className="text-muted text-center"
                  >
                    Sorry no recipe has been added yet, please add to get started
                  </p>
                </Exclamation>
              </div>
            </div>
          </main>,
          <Pagination
            recipesCount={1}
            recipesPagination={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="container"
              >
                <Connect(CategoryButton) />
              </div>,
              <div
                className="container"
              >
                <div
                  style={
                    Object {
                      "marginTop": 100,
                      "textAlign": "center",
                    }
                  }
                >
                  <h4
                    className="container__myrecipes"
                  >
                    All Recipes
                  </h4>
                  <br />
                  <br />
                  <RecipeModal
                    addRecipe={[Function]}
                    recipeCategories={Array []}
                  />
                </div>
                <div
                  className="row"
                >
                  <Exclamation>
                    <p
                      className="text-muted text-center"
                    >
                      Sorry no recipe has been added yet, please add to get started
                    </p>
                  </Exclamation>
                </div>
              </div>,
            ],
            "style": Object {
              "marginTop": 40,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(CategoryButton) />,
                "className": "container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    style={
                      Object {
                        "marginTop": 100,
                        "textAlign": "center",
                      }
                    }
                  >
                    <h4
                      className="container__myrecipes"
                    >
                      All Recipes
                    </h4>
                    <br />
                    <br />
                    <RecipeModal
                      addRecipe={[Function]}
                      recipeCategories={Array []}
                    />
                  </div>,
                  <div
                    className="row"
                  >
                    <Exclamation>
                      <p
                        className="text-muted text-center"
                      >
                        Sorry no recipe has been added yet, please add to get started
                      </p>
                    </Exclamation>
                  </div>,
                ],
                "className": "container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h4
                        className="container__myrecipes"
                      >
                        All Recipes
                      </h4>,
                      <br />,
                      <br />,
                      <RecipeModal
                        addRecipe={[Function]}
                        recipeCategories={Array []}
                      />,
                    ],
                    "style": Object {
                      "marginTop": 100,
                      "textAlign": "center",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "All Recipes",
                        "className": "container__myrecipes",
                      },
                      "ref": null,
                      "rendered": "All Recipes",
                      "type": "h4",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {},
                      "ref": null,
                      "rendered": null,
                      "type": "br",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "addRecipe": [Function],
                        "recipeCategories": Array [],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Exclamation>
                      <p
                        className="text-muted text-center"
                      >
                        Sorry no recipe has been added yet, please add to get started
                      </p>
                    </Exclamation>,
                    "className": "row",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <p
                        className="text-muted text-center"
                      >
                        Sorry no recipe has been added yet, please add to get started
                      </p>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Sorry no recipe has been added yet, please add to get started",
                        "className": "text-muted text-center",
                      },
                      "ref": null,
                      "rendered": "Sorry no recipe has been added yet, please add to get started",
                      "type": "p",
                    },
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "recipesCount": 1,
            "recipesPagination": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
